### DISCUSSION TITLE
What's Your Preferred Method for Automatic Chatting?

### DISCUSSION THREAD

**Discussion by jlplenio** [KevinBytesTheDust/CVAmp#194] [Q&A]
In recent conversations with supporters on [Ko-fi](https://ko-fi.com/jlplenio/posts), a particular question came up:
   What is the optimal mechanism for CTVBot to automatically send chat messages?

Upon startup, the tool reads a `.txt` file where each line corresponds to a unique chat message. What follows next?

#### A. Global Timer with Variability 
- Should a message be sent every 'x' minutes from a randomly chosen user who is currently logged in?
- Or should we use a round-robin approach where all currently logged-in users take turns sending messages?
- Alternatively, should we pick users at random but ensure that each user has sent at least one message before allowing repeats?

#### B. Individual Timers for Each User
- Should each logged-in user operate on their own independent timer, sending messages at intervals of 'x' minutes?

### Additional Considerations:

- What should happen if we reach the end of the chat message list in the `.txt` file?
- Should we shuffle the list of messages upon startup or retain the original sequence?
- Do we need an additional button that would trigger the sending of a chat message from a randomly selected user and message?

**Comment by nathannnnnn12** [comment]
Individual timer or both if u can, as for additional buttons, a emoji button with regular ones and twitch ones for the channels that the bots follow

**Comment by jlplenio** [comment]
Quoted from [ko-fi](https://ko-fi.com/jlplenio/tiers)

> In my opinion there should be 2 options to send messages. 1 that the bots write in order a txt file and 2 write the messages in a random way. Why? because that way we can scrape a twitch chat of a specific category and run that conversation in order and it seems that we are actually talking about that category where we got the chat and the bots do not write random nonsense and it does not seem a real chat. also if you can do that you add randomly to the lines some emoji or emote global twitch so that it is not only the message loaded and thus be less undetectable as a bot. Alternatively, should we pick users at random but ensure that each user has sent at least one message before allowing repeats? this option should also be implemented so that if we connect, for example, 100 tokens, the stream stats will show that you had 100 chatters and thus be sure that all the bots wrote and none of them was missing
> What should happen if we reach the end of the chat message list in the .txt file? that the messages are repeated but now randomly
> In general there should be 2 options, read the text in order and random, the 2 options with its proper timer settings to adjust the time of the messages etc, to be added some emoji or emote twitch to be more real to the 2 options and be more difficult for twitch to detect by algorithm that are bots. users when typing messages in order would not necessarily have to type the tokens in order as well as the messages, you could put that as an additional option to make the chatbot fully customizable
>

**Comment by yomiyo17** [comment]
The bot using Open Ai to actually read chat and create an interaction that emulates actual conversations. I know this would require people to have their own open Ai keys but i dont think this would be an issue for everyone.

**Reply by jlplenio** [reply]
This is the future "cherry on top". Having GPT-chatters which instigate chat conversation and then slowly back off.
